aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,816 @@
+            sender,
+            recipient,
+            amount
+        );
+
+        let bridge_events = borrow_global<BridgeEvents>(@aptos_framework);
+        let bridge_transfer_initiated_events = event::emitted_events_by_handle(
+            &bridge_events.bridge_transfer_initiated_events
+        );   
+        let bridge_transfer_initiated_event = vector::borrow(&bridge_transfer_initiated_events, 0);
+
+        let bridge_transfer_id = bridge_transfer_initiated_event.bridge_transfer_id;
+        
+    }
+
+    #[test(aptos_framework = @aptos_framework, sender = @0xdaff)]
+    #[expected_failure(abort_code = EINVALID_BRIDGE_TRANSFER_ID, location = Self)] // ENOT_FOUND
+    fun test_complete_bridge_with_erroneous_bridge_id_by_relayer(
+        sender: &signer,
+        aptos_framework: &signer
+    ) acquires BridgeEvents {
+        let sender_address = signer::address_of(sender);
+        // Create an account for our recipient
+        native_bridge_core::initialize_for_test(aptos_framework);
+        aptos_account::create_account(sender_address);
+
+        let bridge_transfer_id = b"guessing the id";
+
+        // As relayer I send a complete request and it should fail
+        complete_bridge_transfer(
+            aptos_framework,
+            bridge_transfer_id,
+            valid_eip55(),
+            sender_address,
+            1000,
+            1
+        );
+    }
+}
+
+module aptos_framework::native_bridge_store {
+    use std::bcs;
+    use std::features;
+    use std::vector;
+    use aptos_std::aptos_hash::keccak256;
+    use aptos_std::smart_table;
+    use aptos_std::smart_table::SmartTable;
+    use aptos_framework::ethereum::EthereumAddress;
+    use aptos_framework::system_addresses;
+    use aptos_framework::timestamp;
+    use std::signer;
+    use aptos_framework::timestamp::CurrentTimeMicroseconds;
+
+    friend aptos_framework::native_bridge;
+
+    #[test_only]
+    use std::hash::sha3_256;
+    #[test_only]
+    use aptos_framework::ethereum;
+    #[test_only]
+    use aptos_framework::native_bridge_configuration;
+
+    /// Error codes
+    const EINVALID_PRE_IMAGE : u64 = 0x1;
+    const ENONCE_NOT_FOUND : u64 = 0x2;
+    const EZERO_AMOUNT : u64 = 0x3;
+    const EINVALID_BRIDGE_TRANSFER_ID : u64 = 0x4;
+    const ENATIVE_BRIDGE_NOT_ENABLED : u64 = 0x5;
+    const EINCORRECT_NONCE : u64 = 0x6;
+    const EID_NOT_FOUND : u64 = 0x7;
+
+    const MAX_U64 : u64 = 0xFFFFFFFFFFFFFFFF;
+
+    struct AddressPair<Initiator: store, Recipient: store> has store, copy {
+        initiator: Initiator,
+        recipient: Recipient,
+    }
+
+    /// A smart table wrapper
+    struct SmartTableWrapper<K, V> has key, store {
+        inner: SmartTable<K, V>,
+    }
+
+    /// Details on the outbound transfer
+    struct OutboundTransfer<Initiator: store, Recipient: store> has store, copy {
+        addresses: AddressPair<Initiator, Recipient>,
+        amount: u64,
+        nonce: u64
+    }
+
```

Comment:
New OutboundTransfer should be

```suggestions
struct OutboundTransfer<Initiator: store, Recipient: store> has store, copy {
        bridgeTransferId: vector<u8>,
        addresses: AddressPair<Initiator, Recipient>,
        amount: u64
    }
```

Reply:
Isn't it supposed to be 

```
    /// Details on the outbound transfer
    struct OutboundTransfer<Initiator: store, Recipient: store> has store, copy {
        addresses: AddressPair<Initiator, Recipient>,
        amount: u64,
        nonce: u64
    }
```

because bridge transfer ID can be derived from those details? (initiator, recipient, amount and nonce)